src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/task/createAndRunTaskTool.ts

```diff
@@ -115,13 +115,30 @@ export class CreateAndRunTaskTool implements IToolImpl {
 		if (!outputAndIdle.terminalExecutionIdleBeforeTimeout) {
 			outputAndIdle = await racePollingOrPrompt(
 				() => pollForOutputAndIdle({ getOutput: () => getOutput(terminal), isActive: () => this._isTaskActive(task) }, true, token, this._languageModelsService),
-				() => promptForMorePolling(args.task.label, invocation.context!, this._chatService),
+				() => promptForMorePolling(localize('poll.terminal.waiting', "Continue waiting for `{0}` to finish?", args.task.label), localize('poll.terminal.polling', "Copilot will continue to poll for output to determine when the terminal becomes idle for up to 2 minutes."), invocation.context!, this._chatService),
 				outputAndIdle,
 				token,
 				this._languageModelsService,
 				{ getOutput: () => getOutput(terminal), isActive: () => this._isTaskActive(task) }
 			);
 		}
+		const userInputKind = await getExpectedUserInputKind(outputAndIdle.output);
+		if (userInputKind) {
+			const handleResult = await handleYesNoUserPrompt(
+				userInputKind,
+				invocation.context,
+				this._chatService,
+				terminal,
+				async () => await pollForOutputAndIdle({ getOutput: () => getOutput(terminal), isActive: () => this._isTaskActive(task) }, true, token, this._languageModelsService),
+			);
+			if (handleResult.handled) {
+				if (handleResult.outputAndIdle) {
+					outputAndIdle = handleResult.outputAndIdle;
+				}
+			} else {
+				return { content: [{ kind: 'text', value: `The task is still running and requires user input of ${userInputKind}.` }], toolResultMessage: new MarkdownString(localize('copilotChat.taskRequiresUserInput', 'The task `{0}` is still running and requires user input. {1}', task._label, userInputKind)) };
+			}
+		}
```

Comment:
Sharing this between the 3 tools would be good. This could live in elicitation.ts with the prompt calls too?

Reply:
can't put that in elicitation.ts as there are very specific terminal and task things in this function like outputAndIdle, but moved to bufferOutputPolling